packages/components/src/util/URLResolver.ts

```diff
@@ -34,17 +34,21 @@ export class URLResolver {
         this.mappers = List<URLMapper>([
 
             new ActionMapper('experiment', 'showDataClass', (row, column) => {
-                let url = ['rd', 'dataclass'];
+                let identifier: string;
 
                 // TODO: Deal with junction lookup
-                if (column.has('lookup')) {
-                    url.push(row.get('displayValue').toString());
+                if (row.has('data')) {
+                    // search link doesn't use the same url
+                    identifier = row.getIn(['data', 'name']);
+                } else if (column.has('lookup')) {
+                    identifier = row.get('displayValue').toString();
                 }
                 else {
-                    url.push(row.get('value').toString());
+                    identifier = row.get('value').toString();
+                }
+                if (identifier !== undefined) {
```

Comment:
I find it a bit weird to have return only in the if block, which would result in undefined being returned when evaluated to false (though this might never happen for this function)

Reply:
I agree it is a little weird, just following patterns from other ActionMappers here.  `undefined` is, I believe, the expected result when there isn't a match.